### PR TITLE
Add spark-3.3 profile in building.md

### PR DESCRIPTION
**Pull request by yikf**
### _Why are the changes needed?_

This pr aims to add spark-3.3 profile in building.md document, it's a followup of the https://github.com/apache/incubator-kyuubi/issues/2619

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yikf** [comment]
cc @cfmcgrady

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master

File: docs/develop_tools/building.md [hunk 1]
```diff
@@ -66,6 +66,7 @@ Profile | Default  | Since
 --- | --- | --- 
 -Pspark-3.1 | No | 1.1.0
 -Pspark-3.2 | Yes | 1.4.0
+-Pspark-3.3 | No | 1.6.0
```
Comment (yikf) [inline docs/develop_tools/building.md:69]: left a TODO, we should modify here also if we make spark 3.3 as defualt

**Comment by jiaoqingbo** [comment]
@yikf can you update the company information in the following link？
https://github.com/apache/incubator-kyuubi/discussions/925

**Pull request closed without merging**